src/compiler/checker.ts

```diff
@@ -5831,6 +5839,9 @@ namespace ts {
             if (declaration.kind === SyntaxKind.ExportAssignment) {
                 type = widenTypeForVariableLikeDeclaration(checkExpressionCached((<ExportAssignment>declaration).expression), declaration);
             }
+            else if (isBinaryExpression(declaration)) {

```

Comment:
this case is super suspicious since it seems like it overlaps with the next one, *except* that the next one doesn't support .ts files.

I think you should fix the following case instead of adding this one.